### ISSUE TITLE
Can I finetune the model for a supervised prediction task?

### ISSUE THREAD

**Issue by ReaganGen** [Adibvafa/CodonTransformer#27]
Hi, Thanks for the amazing work! I am trying to add a regression head to your model and finetune the model to be able to predict some property of RNA. Is this suggested, or using logits directly from the model to make the prediction is preferred. Thank you!

**Comment by Adibvafa** [comment]
Hello! Thank you for opening this issue

Could you provide more details on what is the inputs sequence(s) and prediction outcome(s)?

**Comment by ReaganGen** [comment]
Thanks for your quick reply! The input would be RNA sequence, and the prediction outcome would be a single value like the decay rate of the sequence

**Comment by ReaganGen** [comment]
The "using logits directly from the model" means I got the logits for the whole sequence, and calculate the average value of the logits for each amino acid and use this as a prediction value. Generally it is zero-shot

**Comment by gui11aume** [comment]
Thanks @ReaganGen! 
1. CodonTransformer assumes that the input is a coding sequence and it tokenizes the sequence as codons (triplets of nucleotides). An RNA has a 5' and a 3' UTR that are essential for stability so it is not a succession of codons.
2. Codon optimization usually works by facilitating translation, not by making the RNA more stable (but maybe you are of some examples?).
3. The logits of CodonTransformer have dimension 64 (like the codons) whereas RNA stability has dimension 1, like a scalar.

Overall, CodonTransformer does not seem to be the right tool. A simple BigBird for Regression may be more appropriate in my opinion.

**Comment by Adibvafa** [comment]
I agree with @gui11aume 

However, if you wanna give it a try, I suggest passing the coding region of the RNA, taking the last layer embeddings per token, and either averaging them or using the [CLS] token embeddings, feeding that embedding to a regression head
